### PR TITLE
OH Vitals

### PR DESCRIPTION
**Pull request by acrollet**
## Summary

- This work is behind a feature toggle (flipper): YES

In order to meet some deadlines, we are creating an MVP a demo of connecting to a new data source for MHV Medical Records vitals. This is an OCTO initiative to iterate on how we are displaying health data.

- added feature toggle for enablement of new feature
- Created the MVP to use a different data source for vitals on MHV based on a parameter sent by the front end

## Related issue(s)

- follow-on from #19036 
- https://github.com/department-of-veterans-affairs/vets-website/pull/32794

## Testing done

- [x] *New code is covered by unit tests*

Extensive local testing + testing that existing functionality is not impacted on a review instance.

## What areas of the site does it impact?
- MHV vitals endpoint

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by mdewey — DISMISSED** [review]
LGTM

**Review by mmoyer-va — DISMISSED** [review]
Looks good to me!

File: modules/my_health/app/controllers/my_health/v1/vitals_controller.rb [hunk 1]
```diff
@@ -4,7 +4,12 @@ module MyHealth
   module V1
     class VitalsController < MrController
       def index
-        resource = client.list_vitals
+        resource = if params[:from] && params[:to]
+                     client.list_vitals(params[:from], params[:to])
+                   else
+                     client.list_vitals
```
Comment (RachalCassity) [inline modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:10]: list_vitals requires two arguments.

**Review by RachalCassity — CHANGES_REQUESTED** [review]

File: modules/my_health/app/controllers/my_health/v1/vitals_controller.rb (hunk 1 above)
Comment (acrollet) [reply modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:10]: @RachalCassity it only requires arguments when using the Lighthouse Client. Definitely open to a better way of doing this, maybe it should be conditioned on the client class name?

Comment (acrollet) [reply modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:10]: 🤔 or maybe we should make the from and to params optional, the [LH API does not require them](https://developer.va.gov/explore/api/patient-health/docs?version=current)

Comment (RachalCassity) [reply modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:10]: `def list_vitals(from_date: nil, to_date: nil)`

**Review by RachalCassity — CHANGES_REQUESTED** [review]

File: modules/my_health/app/controllers/my_health/v1/vitals_controller.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ module MyHealth
   module V1
     class VitalsController < MrController
       def index
-        resource = client.list_vitals
+        resource = client.list_vitals(params[:from] || nil, params[:to] || nil)
```
[the remark this answers is not in the document]
Comment (acrollet) [reply modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:7]: thanks!

Comment (acrollet) [inline modules/my_health/app/controllers/my_health/v1/vitals_controller.rb:7]:
```suggestion
        resource = client.list_vitals(params[:from], params[:to])
```

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**